### PR TITLE
merge some of my changes into the main github thing

### PR DESCRIPTION
**Pull request by roadworx**
i cleaned up a few files and there's some utilities that i've made that could be useful for the devs going forward

### CODE REVIEW

File: config/InvTweaks.cfg [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #Inventory Tweaks Configuration
 #(Regarding shortcuts, all key names can be found at: http://legacy.lwjgl.org/javadoc/org/lwjgl/input/Keyboard.html)
-#Wed Dec 29 18:46:16 EST 2021
+#Wed May 11 19:57:48 MDT 2022
```
Comment (democat3457) [inline config/InvTweaks.cfg:3]: can you revert this change

File: config/Special_AI.cfg [hunk 2]
```diff
@@ -133,7 +133,7 @@ door_breaking {
...
...rations:carpeted_trapdoor_white, inspirations:carpeted_trapdoor_orange, inspirations:carpeted_trapdoor_magenta, inspirations:carpeted_trapdoor_light_blue, inspirations:carpeted_trapdoor_yellow, inspirations:carpeted_trapdoor_lime, inspirations:carpeted_trapdoor_pink, inspirations:carpeted_trapdoor_gray, inspirations:carpeted_trapdoor_silver, inspirations:carpeted_trapdoor_cyan, inspirations:carpeted_trapdoor_purple, inspirations:carpeted_trapdoor_blue, inspirations:carpeted_trapdoor_brown, inspirations:carpeted_trapdoor_green, inspirations:carpeted_trapdoor_red, inspirations:carpeted_trapdoor_black, netherex:quartz_fence_gate, netherex:nether_brick_fence_gate, netherex:red_nether_brick_fence_gate, netherex:gloomy_nether_brick_fence_gate, netherex:lively_nether_brick_fence_gate, netherex:fiery_nether_brick_fence_gate, netherex:icy_nether_brick_fence_gate, netherex:basalt_fence_gate, netherex:smooth_basalt_fence_gate, netherex:basalt_brick_fence_gate, netherex:basalt_pillar_fence_gate, pyrotech:refractory_door, pyrotech:stone_door, aether_legacy:skyroot_fence_gate, aether_legacy:skyroot_trapdoor, aether_legacy:zanite_trapdoor, aether_legacy:skyroot_door, aether_legacy:zanite_door } ]
```
Comment (democat3457) [inline config/Special_AI.cfg:136]: revert everything in this file

File: config/fantasticlib/reference/enchantments.txt [hunk 3]
```diff
@@ -107,47 +107,47 @@ msmlegacy:stealth
 msmlegacy:extinction
 nyx:lunar_edge
 nyx:lunar_shield
-randomenchantments:momentum
-randomenchantments:breaking
-randomenchantments:butterfingers
-randomenchantments:ricochet
-randomenchantments:teleportation
-randomenchantments:deflect
+randomenchantments:homing
```
Comment (democat3457) [inline config/fantasticlib/reference/enchantments.txt:110]: revert everything in this file

File: config/itlt.cfg [hunk 4]
```diff
@@ -13,7 +13,7 @@ display {
     # Set to true to make minecraft attempt to maximize itself on startup (This is kinda unstable right now, so don't trust it too much)
     B:ShouldMaximizeDisplay=false
 
-    # Set to true to load a custom icon from config/itlt/icon.png
+    # Set to true to load a custom icon from config\itlt\icon.png
```
Comment (democat3457) [inline config/itlt.cfg:16]: revert this

File: config/multimob/modInformation/allBlocks.txt [hunk 5]
```diff
@@ -260,6 +260,7 @@ adpother:sulfur
 adpother:dust
 adpother:coalbed_methane
 adpother:degassing
+adpother:dreadmist
```
Comment (democat3457) [inline config/multimob/modInformation/allBlocks.txt:263]: revert everything in this file and allBlockStates.txt (it wont let me comment on the file because the diffs are too big)

File: scripts/dungeontactics_lootbags.zs [hunk 6]
```diff
@@ -3,47 +3,34 @@ import loottweaker.vanilla.loot.LootTable;
 import loottweaker.vanilla.loot.LootPool;
```
Comment (democat3457) [inline scripts/dungeontactics_lootbags.zs:3]: revert the .zsrc deletion

File: scripts/quark.zs [hunk 7]
```diff
@@ -140,40 +144,32 @@ RecipeBuilder.get("mage")
...
-]);
-recipes.addShapeless("yellow_slimeball",<betterslimes:yellow_slime>*9,[<quark:color_slime:4>]);
+	if(slm[i] != "black") {
+		recipes.addShaped(slm[i] + "_slime_block",qs,[bsa,bsa,bsa]);
+	}
+	recipes.addShapeless(slm[i] + "_slimeball",(bs.items[0])*9,[qs]);
+}
 
-recipes.addShapeless("black_slimeball",<betterslimes:black_slime>*9,[<quark:color_slime:2>]);
+function bluSlm(x as string) as IIngredient {
```
Comment (democat3457) [inline scripts/quark.zs:168]: i dont think you need a function for this, why not just do it inline?

File: scripts/shared/utils/arrayUtils.zs [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+	// trims a string array
+function trimStrArray(array as string, start as int, end as int) as string[] {
+	var newArray = [] as string[];
+	for i in start to end {
+		newArray += array[i];
+	}
+	return newArray;
+}
+
+	// these functions take a string (or two) and adds it onto each member of an array, spitting back out an array of either oredict entries of itemstacks.
```
Comment (democat3457) [inline scripts/shared/utils/arrayUtils.zs:27]:
```suggestion
	// these functions take a string (or two) and adds it onto each member of an array, spitting back out an array of either oredict entries or itemstacks.
```

File: scripts/shared/utils/eventFuncs.zs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+	// checks all the blocks that the player is currently touching
+function nearbyBlockCheck(p as IPlayer, it as string) as bool {
+	if (p.world.getBlock(p.x+-1, p.y, p.z+-1).definition.id == it | // should maybe see if i can combine these two conditions
+	p.world.getBlock(p.x+-1, p.y+1, p.z+-1).definition.id == it) {
+		return true;
+	}
+	return false;
+}
+
+	// goes through armor arrays that contain armor that the player may be currently wearing
```
Comment (democat3457) [inline scripts/shared/utils/eventFuncs.zs:19]: assuming this function is a wip?

File: scripts/shared/utils/recipeUtils.zs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+			print(i.name);
+			new += i;
+		}
+	}
+	
+	return new;
+}
+
+	// removes and hides a whole array. setting "hide" to false means that it only removes, and doesn't also hide
+function removeRecipeArray(array as IItemStack[], hide as bool = true) {
```
Comment (democat3457) [inline scripts/shared/utils/recipeUtils.zs:44]: i thought zenscript doesnt support default parameters?

**Review by democat3457 — COMMENTED** [review]
looks okay, roughly - a lot of files need to be reverted though, since those are auto-generated files (comments have been made regarding those)

the script changes look nice and very useful though

File: .gitignore [hunk 11]
```diff
@@ -117,4 +117,22 @@ config/requious/fluid_emitter.json
...
+config/acronym/saved/material_parts_contenttweaker.json
+config/acronym/saved/saved_sub_blocks_contenttweaker-original.json
+config/acronym/saved/saved_sub_blocks_contenttweaker.json
+config/multimob/modInformation/allBiomeTypes.txt
+config/multimob/modInformation/allBlocks.txt
+config/multimob/modInformation/allBlockStates.txt
+config/multimob/modInformation/allEnchantments.txt
+config/multimob/modInformation/allEntities.txt
+config/multimob/modInformation/allItems.txt
+config/multimob/modInformation/allPotionTypes.txt
```
Comment (democat3457) [inline .gitignore:132]: dont ignore config/multimob

File: .gitignore [hunk 12]
```diff
@@ -117,4 +117,22 @@ config/requious/fluid_emitter.json
...
+config/cd4017be/compiled.dat
+config/deadmanssatchel/DIM\#.example.json
+config/fantasticlib/reference/oredict.txt
```
Comment (democat3457) [inline .gitignore:135]: dont ignore config/fantasticlib/

File: .gitignore [hunk 13]
```diff
@@ -117,4 +117,22 @@ config/requious/fluid_emitter.json
...
+config/fantasticlib/reference/recipes.txt
+config/InvTweaks.cfg
+config/itlt.cfg
```
Comment (democat3457) [inline .gitignore:138]: dont ignore either InvTweaks.cfg or config/itlt.cfg

basically adding a file to the gitignore means the file should never be on the repo - these are important cfg files though, so they should never be in gitignore

**Review by democat3457 — COMMENTED** [review]
looks fine, except for changes to gitignore

**Comment by democat3457** [comment]
also, still address the other comments from the previous review

File: .gitignore (hunk 13 above)
Comment (roadworx) [reply .gitignore]: sorry, it was the only way i could find that'd get the damn things out of the changed files list. they annoy the shit out of me

File: scripts/shared/utils/recipeUtils.zs (hunk 10 above)
Comment (roadworx) [reply scripts/shared/utils/recipeUtils.zs:44]: it does! it also supports functions as parameters, final variables, and custom classes, which is nice :)

unfortunately, it doesn't support private functions, abstracts, or inheritance, which...sucks ass.

**Comment by roadworx** [comment]
that should be everything

File: scripts/shared/utils/recipeUtils.zs (hunk 10 above)
Comment (democat3457) [reply scripts/shared/utils/recipeUtils.zs:44]: funnily enough, all mentions of default parameters say no default parameters - except for the one github issue thread thats adjacent to the mention of default parameter

**Review by democat3457 — APPROVED** [review]
